versions/3.0.md

```diff
@@ -1555,15 +1555,70 @@ description: object created
 
 #### <a name="callbackObject"></a>Callback Object
 
-A container for possible out-of band callbacks from an operation. A callback may be returned from an operation, calling back to the path specified in the operation object.
+A map of possible out-of band callbacks related to the parent operation. Each value in the map is an [operation object](#operationObject) that describes a request that may be initiated by the API provider and the responses expected. The key value used to identify callback object is an expression, that when evaluated at runtime, identifies a URL to used for the callback operation.
 
 ##### Patterned Fields
 Field Pattern | Type | Description
 ---|:---:|---
-<a name="responseName"></a>Callback name | [Callback Operation Object](#operationObject) <span>&#124;</span> [Operation Object](#operationObject) | An operation object used to define a callback payload structure
+<a name="responseName"></a>Callback name | [Callback Operation Object](#operationObject) <span>&#124;</span> [Operation Object](#operationObject) | An operation object used to define a callback request and expected responses
 
 This object can be extended with [Specification Extensions](#specificationExtensions). 
 
+##### Key Expression
+
+The key used to identify the callback object is also an expression that can be evaluated on the runtime HTTP request/response to identify the URL to be used for the callback request.  A simple example might be "$request.body.url".  However, the expression syntax enables accessing the complete HTTP message and reaching into any payload that can map to an object/array/property data model. 
+
+The expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
```

Comment:
Did you consider using [JSON Pointer](https://tools.ietf.org/html/rfc6901) or [JSONPath](http://goessner.net/articles/JsonPath/) for the `body-reference` expressions? In particular, when extracting elements from arrays, it is often necessary to pass in predicates/filters to match a specific array elements. (I'm extrapolating based on from experience with [UnRAVL](https://github.com/sassoftware/unravl/blob/master/doc/Bind.md) which also allows extracting data from response bodies.)